UserInfo is set as Principle but does not satisfy Spring Spec

nenaraab opened the pull request:
@mwdb 
Question: is it possible, that the JWT claim "origin" is not filled? Because in the `UserInfoTestEndUser` test the jwt token had no "origin" claim.

This PR fixes the issue https://github.com/SAP/cloud-security-xsuaa-integration/issues/9

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/cloud-security-xsuaa-integration?pullRequest=10) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/cloud-security-xsuaa-integration?pullRequest=10) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/cloud-security-xsuaa-integration?pullRequest=10) it.</sub>

mwdb commented on spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java:
```diff
@@ -630,6 +631,29 @@ private URI replaceSubdomain(URI uri, String subdomain) {
+			} else {
+				principalName = getLogonName();
+				if (getOrigin() != null) {
+					principalName = getOrigin() + "." + principalName;
```
The "." is a very common seperator. I'd rather use something like /,# or |

nenaraab replied:
@mwdb 
Thanks, Martijn! I've corrected the delimiter to "/"... maybe in future we can make it configurable...

nenaraab commented:
@mwdb 
as discussed I've updated my pull request.
Now, UserInfo does not implement XSPrincipal but UserDetails from spring security.
Furthermore i've refactored UserInfoTestUtil and I've fixed another issue, which was documented here #11 

Can you please review?
